packages/protocol/contracts/common/AddressResolver.sol

```diff
@@ -72,11 +72,8 @@ abstract contract AddressResolver {
     function keyForName(
         uint256 chainId,
         string memory name
-    ) public pure returns (string memory key) {
-        key = string.concat(Strings.toString(chainId), ".", name);
-        // TODO: the next line is cheaper in gas but will break
-        //       many Hardhat tests.
-        // key = string(bytes.concat(bytes32(chainId), bytes(name)));
+    ) public pure virtual returns (string memory) {
+        return string(bytes.concat(bytes32(chainId), bytes(name)));
```

Comment:
Just a follow up on this mechanism:
https://github.com/taikoxyz/taiko-mono/pull/13285#discussion_r1136087299
We need to be sure that off-chain (tests or BE/FE client side typescript) we reconstruct it the same way.

Reply:
We do not use the key offchain in frontend or backend.